application-analytics-ui/src/main/resources/Analytics/Code/Macros/Filtering.xml

```diff
@@ -0,0 +1,364 @@
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </id>
+      <name>
+        <disabled>0</disabled>
+        <name>name</name>
+        <number>2</number>
+        <prettyName>Macro name</prettyName>
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </name>
+      <priority>
+        <disabled>0</disabled>
+        <name>priority</name>
+        <number>11</number>
+        <numberType>integer</numberType>
+        <prettyName>Priority</prettyName>
+        <size>10</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.NumberClass</classType>
+      </priority>
+      <supportsInlineMode>
+        <disabled>0</disabled>
+        <displayFormType>select</displayFormType>
+        <displayType>yesno</displayType>
+        <name>supportsInlineMode</name>
+        <number>5</number>
+        <prettyName>Supports inline mode</prettyName>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.BooleanClass</classType>
+      </supportsInlineMode>
+      <visibility>
+        <cache>0</cache>
+        <disabled>0</disabled>
+        <displayType>select</displayType>
+        <freeText>forbidden</freeText>
+        <largeStorage>0</largeStorage>
+        <multiSelect>0</multiSelect>
+        <name>visibility</name>
+        <number>6</number>
+        <prettyName>Macro visibility</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator>|</separator>
+        <separators>|</separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>Current User|Current Wiki|Global</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </visibility>
+    </class>
+    <property>
+      <async_cached>0</async_cached>
+    </property>
+    <property>
+      <async_context>
+        <value>wiki</value>
+      </async_context>
+    </property>
+    <property>
+      <async_enabled>0</async_enabled>
+    </property>
+    <property>
+      <code>{{velocity}}
+#set ($discard = $xwiki.jsx.use('Analytics.Code.Macros.Filtering'))
+#set ($placeholderStart = '2007-01-01')
+#set ($placeholderEnd = $datetool.get('yyyy-MM-dd'))
+#if($request.analyticsStartDate &amp;&amp; $request.analyticsStartDate != '')
+  #set ($placeholderStart = $request.analyticsStartDate)
+#end
+#if($request.analyticsEndDate &amp;&amp; $request.analyticsEndDate != '')
+  #set ($placeholderEnd = $request.analyticsEndDate)
+#end
+{{html clean="false"}}
+&lt;form id="analyticsFilter" class="form-horizontal"&gt;
+  &lt;div class="form-group"&gt;
+    &lt;!-- Start Date Label --&gt;
+    &lt;label for="analyticsStartDate" class="col-sm-2 control-label"&gt;$services.localization.render('rendering.macro.filtering.startDate')&lt;/label&gt;
```

Comment:
You should add a title for this and the end date label. Maybe get inspired by Matomo 

Also, line exceeds the 120ch limit, below as well